Assets/LeapC/LeapQuaternion.cs

```diff
@@ -0,0 +1,175 @@
+﻿
+/******************************************************************************\
+     * Copyright (C) 2012-2016 Leap Motion, Inc. All rights reserved.               *
+     * Leap Motion proprietary and confidential. Not for distribution.              *
+     * Use subject to the terms of the Leap Motion SDK Agreement available at       *
+     * https://developer.leapmotion.com/sdk_agreement, or another agreement         *
+     * between Leap Motion and you, your company or other organization.             *
+\******************************************************************************/
+namespace Leap
+{
+  using System;
+  using System.Collections.Generic;
+  using System.Runtime.InteropServices;
+
+  /**
+   * The LeapQuaternion struct represents a rotation in three-dimensional space.
+   */
+  public struct LeapQuaternion :
+    IEquatable<LeapQuaternion>
+  {
+    /**
+     * Creates a new LeapQuaternion with the specified component values.
+     */
+    public LeapQuaternion(float x, float y, float z, float w) :
+      this()
+    {
+      this.x = x;
+      this.y = y;
+      this.z = z;
+      this.w = w;
+    }
+
+    /**
+     * Copies the specified LeapQuaternion.
+     */
+    public LeapQuaternion(LeapQuaternion quaternion) :
+      this()
+    {
+      x = quaternion.x;
+      y = quaternion.y;
+      z = quaternion.z;
+      w = quaternion.w;
+    }
+
+    /**
+     * Copies the specified LEAP_QUATERNION.
+     */
+    public LeapQuaternion(LeapInternal.LEAP_QUATERNION quaternion) :
+      this()
+    {
+      x = quaternion.x;
+      y = quaternion.y;
+      z = quaternion.z;
+      w = quaternion.w;
+    }
+
+    /**
+     * Returns a string containing this quaternion in a human readable format: (x, y, z).
+     */
+    public override string ToString()
+    {
+      return "(" + x + ", " + y + ", " + z + ", " + w + ")";
+    }
+
+    /**
+     * Compare LeapQuaternion equality component-wise.
+     */
+    public bool Equals(LeapQuaternion v)
+    {
+      return x.NearlyEquals(v.x) && y.NearlyEquals(v.y) && z.NearlyEquals(v.z) && w.NearlyEquals(v.w);
+    }
+    public override bool Equals(Object obj)
+    {
+      return obj is LeapQuaternion && Equals((LeapQuaternion)obj);
+    }
+
+    /**
+     * Returns true if all of the quaternion's components are finite.  If any
+     * component is NaN or infinite, then this returns false.
+     */
+    public bool IsValid()
+    {
+      return float.IsNaN(x) && float.IsNaN(y) && float.IsNaN(z) && float.IsNaN(w);
```

Comment:
Funny story.  I copied that function from our Vector class which was also wrong.  Both are fixed now.